test/test_nn.py

```diff
@@ -128,6 +128,26 @@ def test_cuda_mask(self):
         unpacked, _ = rnn_utils.pad_packed_sequence(packed)
         self.assertEqual(unpacked.type(), cuda_type_str)
 
+    def test_total_length(self):
+        padded, lengths = self._padded_sequence(torch.FloatTensor)
+        max_length = max(lengths)
+        packed = rnn_utils.pack_padded_sequence(padded, lengths)
+        # test ValueError if total_length < max_length
+        for total_length in (-1, 0, max_length - 1):
+            for batch_first in (True, False):
+                def err_fn():
+                    rnn_utils.pad_packed_sequence(packed, batch_first=batch_first,
+                                                  total_length=total_length)
+            self.assertRaises(ValueError, err_fn)
+        # test that pad_packed_sequence returns results of correct length
+        for total_length_delta in (0, 1, 8):
+            for batch_first in (True, False):
+                total_length = max_length + total_length_delta
+                unpacked, lengths_out = rnn_utils.pad_packed_sequence(packed, batch_first=batch_first,
+                                                                      total_length=total_length)
+                self.assertEqual(lengths, lengths_out)
+                self.assertEqual(unpacked.size(1 if batch_first else 0), total_length)
```

Comment:
Can you please also check that the extra padding is 0 and that the data is still laid out correctly (you can use `pad_packed_sequence` without `total_length`, `cat` and check for equality)?

Reply:
added one below :)